### ISSUE TITLE
ICE with opaque type

### ISSUE THREAD

**Issue by alexkirsz** [rust-lang/rust#85113]
I was playing around with existential types in nightly, forgot to update some code, and hit this ICE.

### Code

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=3794cac3bdc24912ee114d8991502058

```Rust
#![feature(min_type_alias_impl_trait)]
#![feature(impl_trait_in_bindings)]

type OpaqueOutputImpl<'a> = impl Output<'a> + 'a;

trait Output<'a> {}

impl<'a> Output<'a> for &'a str {}

fn cool_fn<'a>(arg: &'a str) -> OpaqueOutputImpl<'a> {
    let out: OpaqueOutputImpl<'a> = arg;
    arg // This was supposed to be `out`
}

fn main() {
    let s = String::from("wassup");
    cool_fn(&s);
}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.54.0-nightly (881c1ac40 2021-05-08) running on x86_64-unknown-linux-gnu
```

### Error output

```
error: internal compiler error: member constraint for `DefId(0:5 ~ playground[b426]::OpaqueOutputImpl::{opaque#0})` has an option region `'_#7r` that is not a universal region
  --> src/main.rs:11:37
   |
11 |     let out: OpaqueOutputImpl<'a> = arg;
   |                                     ^^^
   |
   = note: delayed at compiler/rustc_mir/src/borrow_check/region_infer/mod.rs:697:22

thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', compiler/rustc_errors/src/lib.rs:1019:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.54.0-nightly (881c1ac40 2021-05-08) running on x86_64-unknown-linux-gnu

note: compiler flags: -C embed-bitcode=no -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `playground`

To learn more, run the command again with --verbose.
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
error: internal compiler error: member constraint for `DefId(0:5 ~ playground[b426]::OpaqueOutputImpl::{opaque#0})` has an option region `'_#7r` that is not a universal region
  --> src/main.rs:11:37
   |
11 |     let out: OpaqueOutputImpl<'a> = arg;
   |                                     ^^^
   |
   = note: delayed at compiler/rustc_mir/src/borrow_check/region_infer/mod.rs:697:22

thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', compiler/rustc_errors/src/lib.rs:1019:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/881c1ac408d93bb7adaa3a51dabab9266e82eee8/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/881c1ac408d93bb7adaa3a51dabab9266e82eee8/library/std/src/panicking.rs:435:5
   2: rustc_errors::HandlerInner::flush_delayed
   3: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
   4: core::ptr::drop_in_place<rustc_session::parse::ParseSess>
   5: <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop
   6: core::ptr::drop_in_place<rustc_interface::interface::Compiler>
   7: rustc_span::with_source_map
   8: rustc_interface::interface::create_compiler_and_run
   9: scoped_tls::ScopedKey<T>::set
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</p>
</details>

**Comment by aticu** [comment]
I'm not sure, if it's the same cause, but this ICE seems related (found in #85883 by @Boddlnagg), so I'll post it here instead of in a new issue:

```rust
#![feature(unboxed_closures)]
#![feature(min_type_alias_impl_trait)]
#![feature(impl_trait_in_bindings)]

type Fun = impl Fn<()>;
static BBB: Fun = bbb;
type Out = <Fun as FnOnce<()>>::Output;

fn bbb() {}

fn main() {
    let bar: Out = BBB();
}
```

```text
error: internal compiler error: broken MIR in DefId(0:8 ~ playground[2b78]::main) (NoSolution): failed to normalize `<_ as std::ops::FnOnce<()>>::Output`
  |
  = note: delayed at compiler/rustc_mir/src/borrow_check/type_check/mod.rs:252:27

thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', compiler/rustc_errors/src/lib.rs:1021:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.54.0-nightly (625d5a693 2021-06-01) running on x86_64-unknown-linux-gnu

note: compiler flags: -C embed-bitcode=no -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `playground`

To learn more, run the command again with --verbose.
```

**Comment by JohnTitor** [comment]
The original ICE has been fixed by #86118 but https://github.com/rust-lang/rust/issues/85113#issuecomment-853126040 hasn't, @aticu could you open a new issue if it hasn't been reported yet?
